pluginClientSdkLib/build.gradle

```diff
@@ -78,6 +80,8 @@ uploadArchives {
             }
             // Exclude support.v4, making it an optional dependency
             pom.whenConfigured { pom -> pom.dependencies.removeAll {dep -> dep.groupId == 'com.android.support' && dep.artifactId == 'support-v4' } }
+            // Exclude appcompat-v7, making it an optional dependency
+            pom.whenConfigured { pom -> pom.dependencies.removeAll {dep -> dep.groupId == 'com.android.support' && dep.artifactId == 'appcompat-v7' } }
```

Comment:
I've run gradlew installArchives with the changes I made commented out, and
I am not seeing any evidence of appCompat in the pom file
(.m2\repository\com\twofortyfouram\android-plugin-client-sdk-for-locale\1.0.0\android-plugin-client-sdk-for-locale-1.0.0.pom)
as I would expect to:

<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="
http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.twofortyfouram</groupId>
  <artifactId>android-plugin-client-sdk-for-locale</artifactId>
  <version>1.0.0</version>
  <packaging>aar</packaging>
</project>

I could not find good documentation around pom.whenConfigured, but I read
each line as a directive, not a variable, so I expected that it would run
one after the other, but I could be wrong (and likely am).

On Tue, Feb 24, 2015 at 10:28 AM, Carter Jernigan <notifications@github.com>
wrote:

> In pluginClientSdkLib/build.gradle
> <https://github.com/twofortyfouram/android-plugin-client-sdk-for-locale/pull/1#discussion_r25260285>
> :
>
> > @@ -78,6 +80,8 @@ uploadArchives {
> >              }
> >              // Exclude support.v4, making it an optional dependency
> >              pom.whenConfigured { pom -> pom.dependencies.removeAll {dep -> dep.groupId == 'com.android.support' && dep.artifactId == 'support-v4' } }
> > +            // Exclude appcompat-v7, making it an optional dependency
> > +            pom.whenConfigured { pom -> pom.dependencies.removeAll {dep -> dep.groupId == 'com.android.support' && dep.artifactId == 'appcompat-v7' } }
>
> I think this is broken. The way I read this: we're assigning the
> pom.whenConfigured variable to a closure. Then we re-assign to a different
> closure right after, dereferencing the first closure.
>
> I think the correct solution involves a single closure with dep.groupId
> == 'com.android.support' && (dep.artifactId == 'support-v4' ||
> dep.artifactId == 'appcompat-v7')
>
> One way to test whether this worked as intended is to run ./gradlew
> installArchives and then looking at the local maven repo (on my Mac,
> under ~/m2/). The pom file nested in the maven repo lists the required
> dependencies, and we'd be looking for the absence of support-v4 and
> appcompat-v7.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twofortyfouram/android-plugin-client-sdk-for-locale/pull/1/files#r25260285>
> .
>